src/Microsoft.PowerApps.TestEngine/TestEngine.cs

```diff
@@ -92,6 +92,11 @@ public async Task<string> RunTestAsync(FileInfo testConfigFile, string environme
                     throw new ArgumentNullException(nameof(domain));
                 }
 
+                if (!Directory.Exists(outputDirectory.FullName))
```

Comment:
nit: you could also leverage the`  _fileSystem.Exists((outputDirectory.FullName) ` src\Microsoft.PowerApps.TestEngine\System\FileSystem.cs